themes/developer/developer.theme.sh

```diff
@@ -0,0 +1,229 @@
+#! bash oh-my-bash.module
+
+SCM_THEME_PROMPT_DIRTY=" ${_omb_prompt_brown}✗"
+SCM_THEME_PROMPT_CLEAN=" ${_omb_prompt_bold_green}✓"
+SCM_THEME_PROMPT_PREFIX=" ${_omb_prompt_green}|"
+SCM_THEME_PROMPT_SUFFIX="${_omb_prompt_green}|"
+
+GIT_THEME_PROMPT_DIRTY=" ${_omb_prompt_brown}✗"
+GIT_THEME_PROMPT_CLEAN=" ${_omb_prompt_bold_green}✓"
+GIT_THEME_PROMPT_PREFIX=" ${_omb_prompt_green}|"
+GIT_THEME_PROMPT_SUFFIX="${_omb_prompt_green}|"
+
+RVM_THEME_PROMPT_PREFIX="|"
+RVM_THEME_PROMPT_SUFFIX="|"
+
+function return_delimited {
+	# 1 = flag 2 = tag 3 = value
+	if [[ $1 = "true" ]]; then
+		printf "$2 $3---"
+	elif [[ $1 = "false" ]]; then
+		printf "$2 $3"
+	else
+		printf "$1 $2"
+	fi
+}
+
+function extract_key {
+	value=$(echo "$1" | grep -oP "(?<=$2)[^---]+")
```

Comment:
`grep -P` is not POSIX. We do not require the users to install GNU grep. Please use the POSIX-compatible feature.

Reply:
Just out of curiosity, what does `[^---]` mean? Is it different from `[^-]` in PCRE or perlre?

Reply:
> Just out of curiosity, what does `[^---]` mean? Is it different from `[^-]` in PCRE or perlre?

I again looked at the code. This `[^---]` is probably intended to split the result by `---`, which is inserted by `return_delimited`. However, this doesn't work. `[...]` is a character class, which means that `[^---]` only sees a single character to check if it matches. This construct cannot be used to exclude the string `---`. This is actually equivalent to `[^-]` and matches any character except for `-`. See this:

```console
$ echo 'aa bb-cc---dd ee-ff' | grep -oP '(?<=aa)[^---]+'
 bb
```

It returns ` bb` instead of `bb-cc`. The correct way to exclude `---` is `(?:(?!---).)+`.

```console
$ echo 'aa bb-cc---dd ee-ff' | grep -oP '(?<=aa)(?:(?!---).)+'
 bb-cc
```

Reply:
Also, with this implementation, there is always a space at the beginning. I thought a possibility that this space is intentional so that it separates the element from the previous element, but as far as I see the place it is used, there is already a space, `(`, `)`, etc. to separate the elements. The extra space doesn't seem to be intentional.

Anyway, `grep -P` is not POSIX, and we need to use other way to achieve this. I will change it to use ASCII FS (`$'\x1c'`)to delimit the results.

Reply:
Actually, one doesn't even need to use `$'\x1c'` to delimit the results. One can simply delimit the results with newlines. We do not expect a single prompt element contain newlines.

b4a9b9d